snap/info.go

```diff
@@ -38,45 +38,49 @@ import (
-	Name() string
+	// InstanceName returns the name of the snap.
+	InstanceName() string
 
-	// MountDir returns the base directory of the snap.
-	MountDir() string
+	// StoreName returns the name of the snap as referred to in the store.
+	StoreName() string
 
-	// MountFile returns the path where the snap file that is mounted is installed.
-	MountFile() string
+	// InstanceMountDir returns the base directory of the snap.
+	InstanceMountDir() string
 
-	// HooksDir returns the directory containing the snap's hooks.
-	HooksDir() string
+	// InstanceMountFile returns the path where the snap file that is mounted is installed.
+	InstanceMountFile() string
 
-	// DataDir returns the data directory of the snap.
-	DataDir() string
+	// InstanceHooksDir returns the directory containing the snap's hooks.
+	InstanceHooksDir() string
 
-	// UserDataDir returns the per user data directory of the snap.
-	UserDataDir(home string) string
+	// InstanceDataDir returns the data directory of the snap.
+	InstanceDataDir() string
 
-	// CommonDataDir returns the data directory common across revisions of the snap.
-	CommonDataDir() string
+	// InstanceUserDataDir returns the per user data directory of the snap.
+	InstanceUserDataDir(home string) string
 
-	// UserCommonDataDir returns the per user data directory common across revisions of the snap.
-	UserCommonDataDir(home string) string
+	// InstanceCommonDataDir returns the data directory common across revisions of the snap.
+	InstanceCommonDataDir() string
 
-	// UserXdgRuntimeDir returns the per user XDG_RUNTIME_DIR directory
-	UserXdgRuntimeDir(userID sys.UserID) string
+	// InstanceUserCommonDataDir returns the per user data directory common across revisions of the snap.
+	InstanceUserCommonDataDir(home string) string
 
-	// DataHomeDir returns the a glob that matches all per user data directories of a snap.
-	DataHomeDir() string
+	// InstanceUserXdgRuntimeDir returns the per user XDG_RUNTIME_DIR directory
+	InstanceUserXdgRuntimeDir(userID sys.UserID) string
 
-	// CommonDataHomeDir returns a glob that matches all per user data directories common across revisions of the snap.
-	CommonDataHomeDir() string
+	// InstanceDataHomeDir returns the a glob that matches all per user data directories of a snap.
+	InstanceDataHomeDir() string
 
-	// XdgRuntimeDirs returns a glob that matches all XDG_RUNTIME_DIR directories for all users of the snap.
-	XdgRuntimeDirs() string
+	// InstanceCommonDataHomeDir returns a glob that matches all per user data directories common across revisions of the snap.
+	InstanceCommonDataHomeDir() string
+
+	// InstanceXdgRuntimeDirs returns a glob that matches all XDG_RUNTIME_DIR directories for all users of the snap.
+	InstanceXdgRuntimeDirs() string
 }
 
 // MinimalPlaceInfo returns a PlaceInfo with just the location information for a snap of the given name and revision.
 func MinimalPlaceInfo(name string, revision Revision) PlaceInfo {
+	// TODO consider using instance key
```

Comment:
yes, this will need to use that, it's used to setup a mininal info to drive removals